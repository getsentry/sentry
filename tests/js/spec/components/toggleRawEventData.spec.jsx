import React from 'react';

import {shallow} from 'sentry-test/enzyme';

import EventDataSection from 'app/components/events/eventDataSection';
import KeyValueList from 'app/components/events/interfaces/keyValueList/keyValueList';

const data = {
  metadata: {
    title: 'metadata title',
    type: 'metadata type',
    directive: 'metadata directive',
    uri: 'metadata uri',
    value: 'metadata value',
    message: 'metadata message',
  },
  culprit: 'culprit',
};

describe('EventDataSection', function() {
  const groupData = {
    ...data,
    level: 'error',
    id: 'id',
  };
  const eventData = {
    ...data,
    id: 'id',
    eventID: 'eventID',
    groupID: 'groupID',
    culprit: undefined,
  };
  it('renders formatted', function() {
    const component = shallow(
      <EventDataSection
        group={groupData}
        event={eventData}
        type="extra"
        title="Additional Data"
        raw={false}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('renders raw', function() {
    const component = shallow(
      <EventDataSection
        group={groupData}
        event={eventData}
        type="extra"
        title="Additional Data"
        raw
      />
    );
    expect(component).toMatchSnapshot();
  });
});

describe('KeyValueList', function() {
  const context = {
    somestuff: {andsomeotherstuff: 'here'},
    plussomeotherstuff: 'here',
    andthis: 0,
  };
  const extraDataArray = Object.entries(context);

  it('renders formatted', function() {
    const component = shallow(
      <KeyValueList data={extraDataArray} isContextData raw={false} />
    );

    expect(component).toMatchSnapshot();
  });

  it('renders raw', function() {
    const component = shallow(<KeyValueList data={extraDataArray} isContextData raw />);

    expect(component).toMatchSnapshot();
  });
});
