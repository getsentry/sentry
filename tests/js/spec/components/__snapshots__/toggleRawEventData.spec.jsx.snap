// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDataSection renders formatted 1`] = `
<div
  className=" box"
>
  <div
    className="box-header"
    id="extra"
  >
    <a
      className="permalink"
      href="#extra"
    >
      <em
        className="icon-anchor"
      />
    </a>
    <h3>
      Additional Data
    </h3>
    <div
      className="btn-group pull-right"
    >
      <a
        className="active btn btn-default btn-sm"
        onClick={[Function]}
      >
        Formatted
      </a>
      <a
        className=" btn btn-default btn-sm"
        onClick={[Function]}
      >
        Raw
      </a>
    </div>
  </div>
  <div
    className="box-content with-padding"
  />
</div>
`;

exports[`EventDataSection renders raw 1`] = `
<div
  className=" box"
>
  <div
    className="box-header"
    id="extra"
  >
    <a
      className="permalink"
      href="#extra"
    >
      <em
        className="icon-anchor"
      />
    </a>
    <h3>
      Additional Data
    </h3>
    <div
      className="btn-group pull-right"
    >
      <a
        className=" btn btn-default btn-sm"
        onClick={[Function]}
      >
        Formatted
      </a>
      <a
        className="active btn btn-default btn-sm"
        onClick={[Function]}
      >
        Raw
      </a>
    </div>
  </div>
  <div
    className="box-content with-padding"
  />
</div>
`;

exports[`KeyValueList renders formatted 1`] = `
<table
  className="table key-value"
>
  <tbody>
    <tr
      key="andthis"
    >
      <TableData
        className="key"
        wide={false}
      >
        andthis
      </TableData>
      <td
        className="val"
      >
        <ContextData
          data={0}
        />
      </td>
    </tr>
    <tr
      key="plussomeotherstuff"
    >
      <TableData
        className="key"
        wide={false}
      >
        plussomeotherstuff
      </TableData>
      <td
        className="val"
      >
        <ContextData
          data="here"
        />
      </td>
    </tr>
    <tr
      key="somestuff"
    >
      <TableData
        className="key"
        wide={false}
      >
        somestuff
      </TableData>
      <td
        className="val"
      >
        <ContextData
          data={
            Object {
              "andsomeotherstuff": "here",
            }
          }
        />
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`KeyValueList renders raw 1`] = `
<table
  className="table key-value"
>
  <tbody>
    <tr
      key="andthis"
    >
      <TableData
        className="key"
        wide={false}
      >
        andthis
      </TableData>
      <td
        className="val"
      >
        <ContextData
          data="0"
        />
      </td>
    </tr>
    <tr
      key="plussomeotherstuff"
    >
      <TableData
        className="key"
        wide={false}
      >
        plussomeotherstuff
      </TableData>
      <td
        className="val"
      >
        <ContextData
          data="\\"here\\""
        />
      </td>
    </tr>
    <tr
      key="somestuff"
    >
      <TableData
        className="key"
        wide={false}
      >
        somestuff
      </TableData>
      <td
        className="val"
      >
        <ContextData
          data="{\\"andsomeotherstuff\\":\\"here\\"}"
        />
      </td>
    </tr>
  </tbody>
</table>
`;
